motionplan/tpSpaceRRT_test.go

```diff
@@ -467,76 +446,129 @@ func TestPtgCheckPlan(t *testing.T) {
-	})
-	t.Run("obstacles NOT in world frame cause collision - integration test", func(t *testing.T) {
-		obstacle, err := spatialmath.NewBox(
-			spatialmath.NewPoseFromPoint(r3.Vector{2500, 20, 0}),
-			r3.Vector{10, 2000, 1}, "obstacle",
-		)
-		test.That(t, err, test.ShouldBeNil)
-		geoms := []spatialmath.Geometry{obstacle}
-		gifs := []*referenceframe.GeometriesInFrame{referenceframe.NewGeometriesInFrame(cameraFrame.Name(), geoms)}
-
-		worldState, err := referenceframe.NewWorldState(gifs, nil)
-		test.That(t, err, test.ShouldBeNil)
-
-		err = CheckPlan(ackermanFrame, steps, worldState, fs, startPose, inputs, errorState, logger)
-		test.That(t, err, test.ShouldNotBeNil)
-	})
-	t.Run("checking from partial-plan, ensure success with obstacles - integration test", func(t *testing.T) {
-		// create obstacle behind where we are
-		obstacle, err := spatialmath.NewBox(
-			spatialmath.NewPoseFromPoint(r3.Vector{0, 20, 0}),
-			r3.Vector{10, 200, 1}, "obstacle",
-		)
-		test.That(t, err, test.ShouldBeNil)
-		geoms := []spatialmath.Geometry{obstacle}
-		gifs := []*referenceframe.GeometriesInFrame{referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)}
-
-		worldState, err := referenceframe.NewWorldState(gifs, nil)
-		test.That(t, err, test.ShouldBeNil)
-
-		ov := spatialmath.NewOrientationVector().Degrees()
-		ov.OZ = 1.0000000000000004
-		ov.Theta = -101.42430306111874
-		vector := r3.Vector{669.0803080526971, 234.2834571597409, 0}
-
-		startPose := spatialmath.NewPose(vector, ov)
-
-		err = CheckPlan(ackermanFrame, steps[2:len(steps)-1], worldState, fs, startPose, inputs, errorState, logger)
-		test.That(t, err, test.ShouldBeNil)
-	})
-	t.Run("verify partial plan with non-nil errorState and obstacle", func(t *testing.T) {
-		// create obstacle which is behind where the robot already is, but is on the path it has already traveled
-		obstacle, err := spatialmath.NewBox(
-			spatialmath.NewPoseFromPoint(r3.Vector{0, 0, 0}),
-			r3.Vector{10, 10, 1}, "obstacle",
-		)
-		test.That(t, err, test.ShouldBeNil)
-		geoms := []spatialmath.Geometry{obstacle}
-		gifs := []*referenceframe.GeometriesInFrame{referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)}
-
-		worldState, err := referenceframe.NewWorldState(gifs, nil)
-		test.That(t, err, test.ShouldBeNil)
-
-		errorState := spatialmath.NewPoseFromPoint(r3.Vector{0, 1000, 0})
-		startPose = errorState
-
-		err = CheckPlan(ackermanFrame, steps[2:len(steps)-1], worldState, fs, startPose, inputs, errorState, logger)
-		test.That(t, err, test.ShouldBeNil)
-	})
+	type testCase struct {
+		name             string
+		obstaclesExist   bool
+		obsPosition      r3.Vector
+		obsDims          r3.Vector
+		observerFrame    string
+		errorState       r3.Vector
+		startPosition    r3.Vector
+		startOrientation *spatialmath.OrientationVectorDegrees
+		planIndex        int
+		errorIsNil       bool
+	}
+
+	ov := spatialmath.NewOrientationVector().Degrees()
+	ov.OZ = 1.0000000000000004
```

Comment:
Why

Reply:
this can be changes to just be `1.0` I'll update this.

Reply:
Note that `NewOrientationVector()` returns an `ov` whose `OZ` is already set to 1.